org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverterTestAST8_2.java

```diff
@@ -6582,12 +6583,12 @@ public void test0612() throws JavaModelException {
 			Type type = variableDeclaration.getType();
 			checkSourceRange(type, "boolean[]", contents);
 			assertTrue("Not an array type", type.isArrayType());
-			ArrayType arrayType = (ArrayType) type;
-			Type componentType = componentType(arrayType);
-			assertTrue("Not a primitive type", componentType.isPrimitiveType());
-			PrimitiveType primitiveType = (PrimitiveType) componentType;
-			assertEquals("Not boolean", PrimitiveType.BOOLEAN, primitiveType.getPrimitiveTypeCode());
-			checkSourceRange(primitiveType, "boolean", contents);
+//			ArrayType arrayType = (ArrayType) type;
+//			Type componentType = componentType(arrayType);
+//			assertTrue("Not a primitive type", componentType.isPrimitiveType());
+//			PrimitiveType primitiveType = (PrimitiveType) componentType;
+//			assertEquals("Not boolean", PrimitiveType.BOOLEAN, primitiveType.getPrimitiveTypeCode());
+//			checkSourceRange(primitiveType, "boolean", contents);
```

Comment:
This caught me by surprise so I investigated:
* the test implies to test AST at JLS8
* using source from project `Converter`, `Converter15`, however, selects the AST level from the project.

@srikanth-sankaran , @jarthana  does it make any sense have such tests here? Shouldn't tests always sit in a test class corresponding to the used `Converter*` project, e.g., `ASTConverterAST3Test`?

Reply:
Added TODO